app/models/comment.rb

```diff
@@ -11,44 +11,35 @@ class Comment < ActiveRecord::Base
 
   before_save :escape_body
   before_save :set_mentions
-  after_commit :email_mentioned
 
   def created_by?(user)
     commenter_id == user.id
   end
 
-  # TODO Security? What do you think? Also, should we do this client side too?
-  def escape_body
-    self.body = ERB::Util.html_escape(body)
+  def notify_mentioned_people
+    people_mentioned.each do |mentionee|
+      UserMailer.mention_collaborator(self, mentionee).deliver_later
+    end
   end
 
   private
 
-  def notifier_payload
-    { task_id: task.id, paper_id: task.paper.id }
-  end
-
-  def people_mentioned
-    names = Twitter::Extractor.extract_mentioned_screen_names(body).uniq - [commenter.username]
-    @people_mentioned ||= User.where(username: names)
+  # TODO Security? What do you think? Also, should we do this client side too?
+  def escape_body
+    self.body = ERB::Util.html_escape(body)
   end
 
   # uses the same format as
   # https://dev.twitter.com/overview/api/entities-in-twitter-objects#user_mentions
   def set_mentions
-    self.entities = { user_mentions: [] }
-    people_mentioned.each do |user|
-      handle = '@' + user.username
-      first = body.index(handle)
-      last = first + handle.length
-      indices = { indices: [first, last] }
-      self.entities["user_mentions"] << indices
-    end
+    self.entities = {user_mentions: Twitter::Extractor.extract_mentioned_screen_names_with_indices(body)}
```

Comment:
Line is too long. [105/80]<br>Space inside { missing.<br>Space inside } missing.